core/src/CMakeLists.txt

```
@@ -1,15 +1,29 @@
 #I have to leave these here for tribits
-KOKKOS_INCLUDE_DIRECTORIES(
-  ${CMAKE_CURRENT_BINARY_DIR}
-  ${CMAKE_CURRENT_SOURCE_DIR}
-  ${KOKKOS_TOP_BUILD_DIR}
-)
+IF (KOKKOS_ENABLE_IMPL_DESUL_ATOMICS)
+  find_package(desul_atomics QUIET)
+ENDIF()
+
+IF (KOKKOS_ENABLE_IMPL_DESUL_ATOMICS AND NOT desul_atomics_FOUND)
+  KOKKOS_INCLUDE_DIRECTORIES(
+    ${CMAKE_CURRENT_BINARY_DIR}
+    ${CMAKE_CURRENT_SOURCE_DIR}
+    ${CMAKE_CURRENT_SOURCE_DIR}/desul
+    ${KOKKOS_TOP_BUILD_DIR}
+  )
+ELSE()
+  KOKKOS_INCLUDE_DIRECTORIES(
+    ${CMAKE_CURRENT_BINARY_DIR}
+    ${CMAKE_CURRENT_SOURCE_DIR}
+    ${KOKKOS_TOP_BUILD_DIR}
+  )
+ENDIF()
```

This is confusing to read, it's better to split into two `KOKKOS_INCLUDE_DIRECTORIES` since 3 of the directories are always included.

```suggestion
KOKKOS_INCLUDE_DIRECTORIES(
  ${CMAKE_CURRENT_BINARY_DIR}
  ${CMAKE_CURRENT_SOURCE_DIR}
  ${KOKKOS_TOP_BUILD_DIR}
)

IF (KOKKOS_ENABLE_IMPL_DESUL_ATOMICS AND NOT desul_atomics_FOUND)
  KOKKOS_INCLUDE_DIRECTORIES(
    ${CMAKE_CURRENT_SOURCE_DIR}/desul
  )
ENDIF()
```